dashboard/test/models/section_test.rb

```diff
@@ -1,69 +1,73 @@
 require 'test_helper'
 
 class SectionTest < ActiveSupport::TestCase
-  test "do not attempt to create sections with duplicate random codes" do
-    teacher = create(:teacher)
+  setup do
+    @student = create :student
+    @teacher = create :teacher
+    @section = create :section, teacher: @teacher
 
-    srand 1
-    s1 = Section.create!(user: teacher, name: "section 1")
+    @default_attrs = {user: @teacher, name: 'test-section'}
+  end
```

Comment:
Not necessary, but this can be made more performant by leveraging @wjordan's new `setup_all` feature, to only run setup a single time instead of before each test case. The test cases are still run in rollback transactions, so will remain independent. [Example.](https://github.com/code-dot-org/code-dot-org/blob/d3cf0d522ac5e88dd3344041a4076fb9676dd1ef/dashboard/test/controllers/api/v1/pd/workshops_controller_test.rb#L6-L7)